native-heapdump-example/README.md

```diff
@@ -0,0 +1,130 @@
+# Create a Heap Dump from a Native Executable
+
+You can create a heap dump of a running executable to monitor its execution. Just like any other Java heap dump, it can be opened with the [VisualVM](https://www.graalvm.org/latest/tools/visualvm/) tool. 
+
+For this, you need to add `jvmstat` to the `--enable-monitoring` option (for example, `--enable-monitoring=heapdump,jvmstat`).
+This will allow VisualVM to pick up and list running Native Image processes.
+You can then request heap dumps in the same way you can request them when your application runs on the JVM (for example, right-click on the process, then select "Heap Dump").
+
+1. Create heap dumps with VisualVM.
+2. Dump the initial heap of a native executable using the `-XX:+DumpHeapAndExit` command-line option.
+3. Create heap dumps sending a `SIGUSR1` signal at run time.
+4. Create heap dumps programmatically using the [`org.graalvm.nativeimage.VMRuntime#dumpHeap`](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VMInspectionOptions.java) API.
+
+All approaches are described below.
+
+>Note: By default, heap dumps are created in the current working directory. The `-XX:HeapDumpPath` option can be used to specify an alternative filename or directory. For example:  
+> `./helloworld -XX:HeapDumpPath=$HOME/helloworld.hprof`
+
+>Also note: Creating heap dumps is not available on the Microsoft Windows platform.
+
+## Preparation
+
+1. Download and  install the latest GraalVM JDK with Native Image and VisualVM support using the [GraalVM JDK Downloader](https://github.com/graalvm/graalvm-jdk-downloader):
+    ```bash
+    bash <(curl -sL https://get.graalvm.org/jdk) -c 'visualvm'
+    ```
+
+2. Download the demos repository or clone it as follows:
+    ```bash
+    git clone https://github.com/graalvm/graalvm-demos
+    ```
+    ```bash
+    cd graalvm-demos/native-heapdump-example
```

Comment:
I would call the folder "native-heapdump-**examples**". There are 2 demos shown.